### ISSUE TITLE
attribute _control_char_table from class BaseHTTPRequestHandler is not recognized and reported as "has no attribute"

### ISSUE THREAD

**Issue by jan-spurny** [python/mypy#16920]
**Bug Report**

When subclassing standard library's BaseHTTPRequestHandler, mypy claims it has no attribute `_control_char_table`.

**To Reproduce**

```python
from http.server import BaseHTTPRequestHandler
import io
from unittest.mock import Mock

class MyHandler(BaseHTTPRequestHandler):
    def foo(self) -> None:
        print(len(self._control_char_table))

request = Mock()
request.makefile = Mock(return_value = io.BytesIO(b'some-request'))
h = MyHandler(request, ('127.0.0.1', 123), Mock())
h.foo()
```

In `http.server.BaseHTTPRequestHandler`, `_control_char_table` is defined on these lines:
```python
...
class BaseHTTPRequestHandler(socketserver.StreamRequestHandler):
    ...
    _control_char_table = str.maketrans(
            {c: fr'\x{c:02x}' for c in itertools.chain(range(0x20), range(0x7f,0xa0))})
    _control_char_table[ord('\\')] = r'\\'
    ...
...
```

**Expected Behavior**

1. program prints length of the _control_char_table, which it does:
    ```shell
    $ python http_req.py 
    127.0.0.1 - - [16/Feb/2024 11:08:35] code 400, message Bad request syntax ('some-request')
    127.0.0.1 - - [16/Feb/2024 11:08:35] "some-request" 400 -
    66
    ```
2. mypy doesn't complain about anything

**Actual Behavior**

Mypy complains:
```shell
$ mypy http_req.py 
http_req.py:7: error: "MyHandler" has no attribute "_control_char_table"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```
**Your Environment**

- Mypy version used: mypy 1.8.0
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.11.2
- OS: Debian 12.4
